internal/openshift/openshift_client.go

```diff
@@ -0,0 +1,141 @@
+package openshift
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+	"time"
+)
+
+var logger = log.WithFields(log.Fields{"component": "openshift-client"})
+
+// OpenShift is a client for OpenShift API
+type Client interface {
+	GetRoute(n string, s string) (r string, tls bool, err error)
+}
+
+// OpenShift is a client for OpenShift API
+type OpenShift struct {
+	token  string
+	apiURL string
+	client *http.Client
+}
+
+// NewOpenShift creates new OpenShift client with new http client.
+func NewClient(apiURL string, token string) Client {
+	c := &http.Client{
+		Transport: &http.Transport{
+			MaxIdleConnsPerHost: 20,
+		},
+		Timeout: time.Duration(10) * time.Second,
+	}
+
+	return NewClientWithHttp(c, apiURL, token)
+}
+
+// NewOpenShiftWithClient create new OpenShift client with given HTTP client.
+func NewClientWithHttp(client *http.Client, apiURL string, token string) Client {
+	if !strings.HasPrefix(apiURL, "http") {
+		apiURL = fmt.Sprintf("https://%s", strings.TrimRight(apiURL, "/"))
+	}
+	return &OpenShift{
+		apiURL: apiURL,
+		token:  token,
+		client: client,
+	}
+}
+
+// GetRoute returns the OpenShift route for a given namespace and service and whether the route uses TLS.
+func (o *OpenShift) GetRoute(n string, s string) (string, bool, error) {
```

Comment:
@vpavlin, on second thought. This might not be enoug.h. We moved this code from Idler to here, but the problem is still that GetRoute for a user in a different cluster won't resolve. There is more needed here. 

If we were to know the cluster URL of the namespace/user, we could adopt GetRoute to target the right cluster, but I am not sure how to do this. Effectively we will need to already partly implement some multi-cluster support :-(

Reply:
As a super ugly workaround, we could for now manually construct the URL for the new cluster. AFAIU, this URL is pretty much static and it is just the namespace which differs. We could call GetRoute and if it does not return anything, we assume it is a user from the new cluster and construct the route accordingly.

I am even wondering do we really need in the long run to determine the route dynamically? It is more flexible in general, but would it be not enough to determine the target cluster and then build the URL by combining namespace and static cluster URL part?

Reply:
Or we somehow use this OSO Proxy, but I am not sure what it does and how to use it.

Reply:
Right, this is not enough, we need to be able pass OSO api url to GetRoute call.

We can determine cluster url from GetTenantInfo call which is used in both flows (webhook and UI) - we already store the cluster url in ProxyCacheItem structure

Reply:
ahh, cool. Seems you have a plan :-) What's about the token? Is it the same for all clusters? Is it enough to just pass the OSO API URL and reuse the token? Thinking back to our conversation yesterday, I think that's what you always said, but I did not quite understand ;-)

Reply:
So the interface would become GetRoute(osoURL string, namespace string, service string), right? Assuming the token is the same and we do already know which cluster a tenant is provisioned on, it should indeed be easy. I guess we also remove _apiURL_ from the OpenShift struct.

Sounds like a plan. I might be able to work today, but we can sync later. If you want to get started, go ahead. I also need to do several things first. I will check-in with you first before doing anything more on this.